### PR TITLE
dot frontend

### PR DESCRIPTION
**Pull request by abou704**
I added the function dot for the frontend API
issue #3287

### CODE REVIEW

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 1]
```diff
@@ -6,3 +6,8 @@ def det(input, name=None):
 
 
 det.unsupported_dtypes = ("float16", "bfloat16")
+
+
+def tensordot(x, y, axes, name=None):
+	return ivy.tensordot(x, y, axes=axes)
```
Comment (ankitdsi2010) [inline ivy/functional/frontends/tensorflow/linalg.py:12]: properly indent by using a single tab space

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 2]
```diff
@@ -22,6 +23,7 @@ def _get_dtype_and_matrix(draw):
     )
 
 
+#det
```
Comment (ankitdsi2010) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:26]: replace with `# det`

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 3]
```diff
@@ -45,3 +47,32 @@ def test_tensorflow_det(
         fn_tree="linalg.det",
         input=np.asarray(x, dtype=input_dtype),
     )
+
+
+#tensordot v1
```
Comment (ankitdsi2010) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:52]: Replace with `# tensordot`

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 4]
```diff
@@ -45,3 +47,32 @@ def test_tensorflow_det(
...
+@given(
+    dtype_and_inputs_axes=test_linalg._get_dtype_value1_value2_axis_for_tensordot(ivy_tf.valid_float_dtypes),
```
Comment (ankitdsi2010) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:54]: line is too long, try to split it into 2 lines

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py [hunk 5]
```diff
@@ -45,3 +47,32 @@ def test_tensorflow_det(
...
+    as_variable=st.booleans(),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.tensorflow.tensordot"
+    ),
+    native_array=st.booleans(),
+)
+def test_tensorflow_tensordot(
+    dtype_and_inputs_axes, as_variable, num_positional_args, native_array, fw
+):
+    dtype, x, y, axes= dtype_and_inputs_axes
```
Comment (ankitdsi2010) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:64]: there should be single space both to the left and right of `=`

**Review by ankitdsi2010 — CHANGES_REQUESTED** [review]

**Review by ankitdsi2010 — CHANGES_REQUESTED** [review]
Resolve why the tensordot tests are failing in the 'test-frontend-tensorflow / run-nightly-tests' as given [here](https://github.com/unifyai/ivy/runs/7938301689?check_suite_focus=true#step:4:142)

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py (hunk 5 above)
Comment (abou704) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py:64]: Thank you for your help. I'll change that

**Pull request merged**